src/System.Private.CoreLib/src/System/Diagnostics/Eventing/RuntimeEventSource.cs

```diff
@@ -54,6 +60,12 @@ protected override void OnEventCommand(EventCommandEventArgs command)
                 _threadPoolThreadCounter = _threadPoolThreadCounter ?? new PollingCounter("threadpool-thread-count", this, () => ThreadPool.ThreadCount) { DisplayName = "ThreadPool Thread Count" };
                 _monitorContentionCounter = _monitorContentionCounter ?? new IncrementingPollingCounter("monitor-lock-contention-count", this, () => Monitor.LockContentionCount) { DisplayName = "Monitor Lock Contention Count", DisplayRateTimeScale = new TimeSpan(0, 0, 1) }; 
                 _threadPoolQueueCounter = _threadPoolQueueCounter ?? new PollingCounter("threadpool-queue-length", this, () => ThreadPool.PendingWorkItemCount) { DisplayName = "ThreadPool Queue Length" };
+                _gcTimeCounter = _gcTimeCounter ?? new PollingCounter("time-in-gc", this, () => GC.GetPercentTimeInGC()) { DisplayName = "Time in GC" };
+                _gen0SizeCounter = _gen0SizeCounter ?? new PollingCounter("gen-0-size", this, () => GC.GetGenerationSize(0)) { DisplayName = "Gen 0 Size" };
```

Comment:
is there actually any chance that you would need to call them separately? if you need to get all of these each time you need the counter value, why not just make one call and get all of the info instead of getting them separately?

Reply:
I had thought of doing that previously but the main rationale behind this was the hope that we would be able to some day expose this as a public managed API that users can use without explicitly consuming the counters. Our goal is that we would be able to have a corresponding managed API that users can invoke directly to get the metrics exposed by every runtime counter that we have. I wasn't pursuing this goal that aggressively at first but it seems like I had misunderstood @noahfalk and @tommcdon and this is somewhat important.

Would you be okay with exposing these internal only APIs as a public API? Please feel free to grab me offline to chat more about it if you have any concerns :-)

Reply:
Why are these internal?